### PR TITLE
[draw API] Avoid some overhead in font.draw_glyph_with_pen()

### PR DESCRIPTION
**Pull request by justvanrossum**
To draw with a pen, don't go via the Python DrawFuncs layer, but use C callbacks directly.

When using a RecordingPen, this is about 25-30% faster.

I mostly wrote this to learn more about the draw API. It adds some lower level code for relatively little gain. I do think it's an improvement overall, but won't complain if it gets rejected.

### CODE REVIEW

File: src/uharfbuzz/_harfbuzz.pyx [hunk 1]
```diff
@@ -546,24 +546,61 @@ cdef class Font:
...
-        funcs.set_close_path_func(close_path, pen)
-        funcs.get_glyph_shape(self, gid)
+        cdef hb_draw_funcs_t* drawfuncs = hb_draw_funcs_create()
+        try:
+            hb_draw_funcs_set_move_to_func(drawfuncs, _pen_move_to_func, NULL, NULL)
+            hb_draw_funcs_set_line_to_func(drawfuncs, _pen_line_to_func, NULL, NULL)
+            hb_draw_funcs_set_cubic_to_func(drawfuncs, _pen_cubic_to_func, NULL, NULL)
+            hb_draw_funcs_set_quadratic_to_func(drawfuncs, _pen_quadratic_to_func, NULL, NULL)
+            hb_draw_funcs_set_close_path_func(drawfuncs, _pen_close_path_func, NULL, NULL)
+            hb_font_get_glyph_shape(self._hb_font, gid, drawfuncs, <void*>pen)
```
Comment (khaledhosny) [inline src/uharfbuzz/_harfbuzz.pyx:556]: While at it, can we set the draw functions once and reuse them?

Comment (justvanrossum) [reply src/uharfbuzz/_harfbuzz.pyx:556]: We could, but since this it at the C level, I don't think this will help much. It's one allocation and 5 C calls.

Comment (khaledhosny) [reply src/uharfbuzz/_harfbuzz.pyx:556]: One `calloc`/`free` call per glyph, these things add up (I’m kidding, that is probably so tiny to be measurable anyway).

Comment (justvanrossum) [reply src/uharfbuzz/_harfbuzz.pyx:556]: I bet a no-op Python moveTo call is at least a magnitude more expensive :)

**Review by khaledhosny — APPROVED** [review]
Thanks!  25-30% faster is not nothing.

**Pull request merged**